### DISCUSSION TITLE
How to extend diagonal edge of pixels towards the bottom edge of image?

### DISCUSSION THREAD

**Discussion by ghost** [ImageMagick/ImageMagick#6349] [Help] [answered]
### ImageMagick version

7.0.8-Q16-HDRI

### Operating system, version and so on

Windows 10

### Description

Hi.

My goal is to extend the exact diagonal edge of pixels to the bottom edge of the image. I attached a sample input and output image to illustrate. I did this freehand so its not exact but hopefully its enough to communicate the idea. I know its kinda weird but its part of a series of effects I am applying to an image to achieve a particular psychedelic look I am going for and I am stuck on this step.  My initial thought was to somehow use -interpolation edge but I am not dealing with virtual pixels so that wont work. Any help is appreciated!

![a](https://github.com/ImageMagick/ImageMagick/assets/80292941/83777eed-e9c9-4f50-826b-bb1ab07d28ab)
![b](https://github.com/ImageMagick/ImageMagick/assets/80292941/5bc8a6cf-d1ef-43b3-a843-898d9833b29e)

**Comment by snibgo** [comment, marked as answer]
> My goal is to extend the exact diagonal edge of pixels ...

The exact diagonal pixels are white. To get near your desired output, we should not duplicate the diagonal. Is this a bug in whatever made the input image?

This Windows BAT script gives roughly your desired output.
```
magick ^
  c:\web\im\diagLines.png -write mpr:INP ^
  -set option:WW %%w ^
  -set option:HH %%h ^
  ( +clone ^
    -fill Black -colorize 100 ^
    -stroke White -draw "line 4,0 %%[fx:w-1],%%[fx:h-5]" ^
  ) ^
  -alpha off ^
  -compose CopyOpacity -composite ^
  -scale "x1^!" ^
  -alpha off ^
  -scale "%%[WW]x%%[HH]^!" ^
  ( +clone ^
    -fill White -colorize 100 ^
    -fill Black -draw "polygon 4,0 %%[fx:w-1],0 %%[fx:w-1],%%[fx:h-5] " ^
  ) ^
  -alpha off ^
  -compose CopyOpacity -composite ^
  mpr:INP ^
  -compose DstOver -composite ^
  x.png
```

**Reply by ghost** [reply]
Yes, I have modified the image so that the diagonal is not white. Its not a bug its just an error in my design. But the actual input images I will have diagonal that is not white.

Thank you so much for the script!

Its going to take me a while to understand it but I sort of know basically what its doing. Im working on implementing it now.
